### PR TITLE
Add driver for hx711 weight scale ADC

### PR DESCRIPTION
**Pull request by mlyszczek**
Patch adds driver to support hx711 adc designed for weigh scales. Example initialization is added in stm32f411-minimum board.

Zero impact on other code.

### CODE REVIEW

**Comment by acassis** [comment]
Please fix this warning/error:

/home/runner/work/nuttx/nuttx/Documentation/components/drivers/special/hx711.rst:document isn't included in any toctree

**Comment by mlyszczek** [comment]
Fixed that, and moved that driver from special/ to character/ since it's just a better match.

**Comment by acassis** [comment]
@mlyszczek thank you! Actually I think you could move it from analog/ to sensors/ because it is more a sensor than an analog device

**Comment by acassis** [comment]
No, I think you are right, the datasheet said it is a sensor: https://cdn.sparkfun.com/datasheets/Sensors/ForceFlex/hx711_english.pdf

**Comment by mlyszczek** [comment]
> @mlyszczek thank you! Actually I think you could move it from analog/ to sensors/ because it is more a sensor than an analog device

Hmm... I think it may be closer to analog? After all, hx711 is just an ADC with some specialty. On the other hand, because of that specialty it makes it closer to sensor? But in theory it also could be used as an ADC - like connecting tensometer to sim racing pedal, and read pedal position that way. Driver can indeed work with that as well.

File: boards/arm/stm32/stm32f411-minimum/src/stm32_hx711.c [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  .cleanup = NULL
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int stm32_hx711_clock_set(unsigned char minor, int value)
+{
+  (void)minor;
```
Comment (acassis) [inline boards/arm/stm32/stm32f411-minimum/src/stm32_hx711.c:61]: Why to do that? Sorry my ignorance, but I really don't know why is it necessary?

Comment (mlyszczek) [reply boards/arm/stm32/stm32f411-minimum/src/stm32_hx711.c:61]: No need to apologize. (void)variable_name, silents compiler warning that variable is unused. And in this case only one hx711 is supported so minor is not used. Now that I write it I think we have UNUSED() macro somewhere that does exactly this. Will change it.

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 any idea of this error:
> 
> ```
> make: *** [tools/Unix.mk:689: olddefconfig] Error 1
>   Enabling CONFIG_ARMV7M_TOOLCHAIN_GNU_EABI
> make: *** [tools/Unix.mk:689: olddefconfig] Error 1
>   Building NuttX...
>     Checking NuttX by Codechecker...
> /github/workspace/sources/nuttx/tools/testbuild.sh: line 252: CodeChecker: command not found
>     Storing analysis result to CodeChecker...
>       Generating HTML report...
> /github/workspace/sources/nuttx/tools/testbuild.sh: line 256: CodeChecker: command not found
>       Compressing logs...
> /github/workspace/sources/nuttx/tools/testbuild.sh: line 370: /github/workspace/sources/nuttx/../nuttx/nuttx.manifest: No such file or directory
>   Normalize sama5d4-ek/nxwm
> make: *** [tools/Unix.mk:689: olddefconfig] Error 1
> make: *** [tools/Unix.mk:712: savedefconfig] Error 1
> make: *** [tools/Unix.mk:689: olddefconfig] Error 1
> ```
> 
> @xiaoxiang781216 I think we need to think more about this CI and how to improve it. Espressif implemented internally a CI based on Robot CI (AFAIK). Maybe @tmedicci could give more details.

I saw that the broken of codechecker is more often than other, @W-Mai, could you disable codechecker ci? It's fine since codechecker report many false alarm and ci never enforce it.

**Comment by hartmannathan** [comment]
@mlyszczek there seems to be a problem with this PR...

boards/arm/stm32/stm32f411-minimum/Kconfig contains:

```
if STM32F411MINIMUM_GPIO
source boards/arm/stm32/stm32f411-minimum/Kconfig.gpio
endif
```

But that Kconfig.gpio file doesn't exist, so all builds break with:

```
boards/arm/stm32/stm32f411-minimum/Kconfig:83: can't open file "boards/arm/stm32/stm32f411-minimum/Kconfig.gpio"
make: *** [tools/Unix.mk:689: olddefconfig] Error 1
ERROR: failed to refresh
```

At least for me...

**Comment by mlyszczek** [comment]
@hartmannathan yes, it's a leak from another feature I'm working on for black pill, sorry :( I'll make a quick patch now.

**Pull request merged**